src/http-api/resources/floodsub.js

```diff
@@ -0,0 +1,87 @@
+'use strict'
+
+const debug = require('debug')
+const log = debug('http-api:floodsub')
+log.error = debug('http-api:floodsub:error')
+
+exports = module.exports
+
+exports.start = {
+  handler: (request, reply) => {
+    request.server.app.ipfs.floodsub.start((err, floodsub) => {
+      if (err) {
+        log.error(err)
+        return reply({
+          Message: `Failed to start: ${err}`,
+          Code: 0
+        }).code(500)
+      }
+
+      return reply(floodsub)
+    })
+  }
+}
+
+exports.subscribe = {
+  handler: (request, reply) => {
+    const discover = request.query.discover || null
+    const topic = request.params.topic
+
+    request.server.app.ipfs.floodsub.subscribe(topic, { discover }, (err, stream) => {
+      if (err) {
+        log.error(err)
+        return reply({
+          Message: `Failed to subscribe to topic ${topic}: ${err}`,
+          Code: 0
+        }).code(500)
+      }
+
+      // hapi is not very clever and throws if no
+      // - _read method
+      // - _readableState object
+      // are there :(
+      if (!stream._read) {
+        stream._read = () => {}
+        stream._readableState = {}
+      }
+      return reply(stream)
```

Comment:
I have a hunch that this won't work. This stream of objects needs to be converted to ndjson stream